src/main/java/org/wso2/carbon/module/swiftiso20022/utils/JsonToMt940Utils.java

```diff
@@ -0,0 +1,332 @@
+     * @return Constructed statement lines
+     */
+    private static List<String> constructStatementLines(RequestPayloadModel requestPayload) {
+        List<TransactionModel> transactions = requestPayload.getTransactions();
+        List<String> statementLines = new ArrayList<>();
+
+        StatementLineModel statementLineModel;
+        for (TransactionModel transaction : transactions) {
+            statementLineModel = new StatementLineModel();
+            statementLineModel.setValueDate(transaction.getDateTime());
+            statementLineModel.setEntryDate(transaction.getDateTime().substring(4));
+            statementLineModel.setTransactionIndicator(transaction.getTransactionIndicator());
+            statementLineModel.setFundCode(transaction.getCurrency().substring(2));
+            statementLineModel.setAmount(formatAmountValue(transaction.getAmount()));
+            statementLineModel.setTransactionType(transaction.getTransactionType());
+            statementLineModel.setCustomerReference(transaction.getCustomerReference());
+            statementLineModel.setBankReference("//" + transaction.getTransactionReference());
+            if (transaction.getSupplementaryData() != null) {
+                statementLineModel.setSupplementaryData("\n" + transaction.getSupplementaryData());
+            }
+            if (transaction.getInformation() != null) {
+                statementLineModel.setInformation("\n:86:" + transaction.getInformation());
+            }
+
+            statementLines.add(statementLineModel.toString());
+        }
+
+        return statementLines;
+    }
+
+    /**
+     * Method to construct the statement Number.
+     *
+     * @return Statement Number
+     */
+    private static String constructStatementNumber(RequestPayloadModel requestPayload) {
+        DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+        Date date = new Date();
+        String dateTime = formatter.format(date);
+        long diffInDays = calculateDaysDiff(dateTime);
+        return Long.toString(diffInDays).concat("/" + requestPayload.getSequenceNumber());
+    }
+
+    /**
+     * Method to format the amount value by removing commas and replacing dots with commas.
+     *
+     * @param amount  Amount Value
+     * @return       Formatted Amount Value
+     */
+    private static String formatAmountValue(String amount) {
+        String formattedAmount = amount.replace(",", "").replace(".", ",");
+        if (formattedAmount.contains(",")) {
+            return formattedAmount;
+        } else {
+            return formattedAmount + ",00";
+        }
+    }
+
+    /**
+     * Calculate the difference between the current date and a given date.
+     *
+     * @param endDate   End Date
+     * @return Difference in days
+     */
+    private static long calculateDaysDiff(String endDate) {
+        LocalDate end = LocalDate.parse(endDate);
+        String startDate = Integer.toString(end.getYear()).concat("-01-01");
```

Comment:
Are we calculating difference between the current date and a given date, or number of days from jan 1st of that year right?

Reply:
Calculating the number of days from Jan 1st of that year